lightning/src/ln/channel.rs

```diff
@@ -1076,8 +1083,14 @@ impl<Signer: Sign> Channel<Signer> {
 		}
 
 		// Check sanity of message fields:
-		if msg.funding_satoshis >= MAX_FUNDING_SATOSHIS {
-			return Err(ChannelError::Close(format!("Funding must be smaller than {}. It was {}", MAX_FUNDING_SATOSHIS, msg.funding_satoshis)));
+		if !their_features.supports_wumbo() && msg.funding_satoshis >= MAX_FUNDING_SATOSHIS_NO_WUMBO {
+			return Err(ChannelError::Close(format!("Funding must be smaller than {}. It was {}", MAX_FUNDING_SATOSHIS_NO_WUMBO, msg.funding_satoshis)));
+		}
+		if msg.funding_satoshis >= config.peer_channel_config_limits.max_funding_satoshis {
+			return Err(ChannelError::Close(format!("Per our config, funding must be smaller than {}. It was {}", config.peer_channel_config_limits.max_funding_satoshis, msg.funding_satoshis)));
+		}
+		if msg.funding_satoshis > TOTAL_BITCOIN_SUPPLY_SATOSHIS {
```

Comment:
Note, here the check is superior only, when above it's superior-or-equal. I don't think it changes that much in practice but..